### ISSUE TITLE
"Optimise Vertex Buffers For Shaders" breaks smooth shaded faces

### ISSUE THREAD

**Issue by Slamy** [OGRECave/ogre-next#202]
I first thought that this was an ogre bug. But it seems that this converter was to blame.
Please have a look at this model.
![blender](https://user-images.githubusercontent.com/5560415/120931880-bfabec00-c6f3-11eb-99da-d926578324b1.PNG)

I've exported it using "Optimise Vertex Buffers For Shaders" checked and opened it in the HLMS editor.
![WithShaderOptimize](https://user-images.githubusercontent.com/5560415/120931896-d6524300-c6f3-11eb-877e-367cdc064103.PNG)
The specular part looks weird and divided. Then I disabled this option and got this.
![WithoutShaderOptimize](https://user-images.githubusercontent.com/5560415/120931915-ed913080-c6f3-11eb-84bd-959c5829025a.PNG)

I don't reallly know why this happens and how. But still I want you to know this.

**Comment by darksylinc** [comment]
Optimization is lossy.

If the mesh is too big (scale problem, e.g. size > 50 units) precision problems appear in the position due to half floating point

Same happens for UVs. If UV are outside 0...1 range half floating point may not be accurate enough.

QTangent optimization is the less prone to problems, but I suppose if the topology is funny it could cause trouble specially if the normals or tangents perform near 180° flips across edges

The first thing to do would be to enable each optimization one by one to find the culprit. Once isolated it may be possible to fix by changing the mesh. But depending on the cause, the solution may be to disable that specific flag, because basically it'd be "working as intended" and the optimization simply should not be applied for this case

**Comment by Slamy** [comment]
I mean. Now I know and knowing is half the battle. But how shall this tool proceed from here? I now know this but the next poor soul will fall into this trap again. Concerning psychology the usual user will always click on something which has the term "Optimize" in it.

@paroj There should be at least some sort of warning in the blender2ogre tool next to the check box. Or even rename it to "experimental optimization".

**Comment by darksylinc** [comment]
I agree on the wording/messaging. But "Experimental" would send the wrong message (it's not experimental, it's not going to improve over time).

The technically correct term would be "lossy optimization" or "Optimization (lossy)", but I don't know if it's user friendly enough.
